app/models/scoped_event_feed.rb

```diff
@@ -0,0 +1,38 @@
+class ScopedEventFeed
+  attr_reader :root, :event_subjects, :page, :per
+  def initialize(root_type, root_id, page = 0, per = 20)
+    @root = root_type.classify.constantize.find(root_id)
+    @event_subjects = EventHierarchy.new(root).children
+    @page = page
+    @per = per
+  end
+
+  def events
+    if @events
+      @events
+    else
+      event_ids = event_subjects.group_by { |x| x.class.to_s }.inject([]) do |ids, (type, vals)|
+        ids + Event.where(subject_type: type, subject_id: vals.map(&:id)).pluck(:id)
+      end
+      @events = Event.includes(:subject, originating_user: [:organization])
+        .where(id: event_ids)
+        .order('events.created_at desc')
+        .page(page)
+        .per(per)
+    end
+  end
+
+  def comments
+    if @comments
+      @comments
+    else
+      comment_ids = events.select { |e| e.state_params['comment'].present? }
```

Comment:
Unfortunately the `subject` of a `commented on` event is the thing being commented on, not the comment itself. That means we're doing some gymnastics here to grab all the comments mentioned in any of the events in our feed so that @jmcmichael can display them inline (a la Github). We're basically just building up a hash of comment_id -> comment object so we can look them up quickly when generating the json.